test/calc_data.test.js

```diff
@@ -0,0 +1,213 @@
+            nft_addr: "nft1",
+            percent: "0.75"
+        },
+        {
+            LP_addr: "lp2",
+            nft_addr: "nft2",
+            percent: "0.35"
+        }
+    ]
+
+    const vebals = [
+        {
+            LP_addr: "lp1",
+            balance: "100",
+            locked_amt: "50"
+        },
+        {
+            LP_addr: "lp2",
+            balance: "200",
+            locked_amt: "75"
+        }
+    ]
+
+    const vebals_realtime = [
+        {
+            LP_addr: "lp1",
+            balance: "120",
+            locked_amt: "60"
+        },
+        {
+            LP_addr: "lp2",
+            balance: "220",
+            locked_amt: "85"
+        }
+    ]
+
+    const nftinfo = [
+        {
+            nft_addr: "nft1",
+            owner_addr: "lp1"
+        },
+        {
+            nft_addr: "nft2",
+            owner_addr: "lp2"
+        }
+    ]
+
+    const { nftinfo: resultNftinfo, allocations: resultAllocations } =
+        calculateAllocations({
+            allocations,
+            allocations_realtime,
+            vebals,
+            vebals_realtime,
+            nftinfo
+        })
+
+    const expectedNftinfo = [
+        {
+            nft_addr: "nft1",
+            owner_addr: "lp1",
+            ve_allocated: 100,
+            ve_allocated_realtime: 180,
+            ocean_allocated: 25,
+            ocean_allocated_realtime: 45,
+            ve_allocated_owner: 100,
+            ve_allocated_realtime_owner: 180,
+            ocean_allocated_owner: 25,
+            ocean_allocated_realtime_owner: 45
+        },
+        {
+            nft_addr: "nft2",
+            owner_addr: "lp2",
+            ve_allocated: 100,
+            ve_allocated_realtime: 154,
+            ocean_allocated: 18.75,
+            ocean_allocated_realtime: 29.75,
+            ve_allocated_owner: 100,
+            ve_allocated_realtime_owner: 154,
+            ocean_allocated_owner: 18.75,
+            ocean_allocated_realtime_owner: 29.75
+        }
+    ]
+
+    const expectedAllocations = [
+        {
+            LP_addr: "lp1",
+            nft_addr: "nft1",
+            percent: "0.5",
+            ve_amt: 100,
+            ocean_amt: 25
+        },
+        {
+            LP_addr: "lp2",
+            nft_addr: "nft2",
+            percent: "0.25",
+            ve_amt: 100,
+            ocean_amt: 18.75
+        }
+    ]
+
+    expect(resultNftinfo[0].ocean_allocated_realtime).toBeCloseTo(
+        expectedNftinfo[0].ocean_allocated_realtime,
+        5
+    )
+    expect(resultNftinfo[0].ocean_allocated_realtime_owner).toBeCloseTo(
+        expectedNftinfo[0].ocean_allocated_realtime_owner,
+        5
+    )
+    expect(resultNftinfo[1].ocean_allocated_realtime).toBeCloseTo(
+        expectedNftinfo[1].ocean_allocated_realtime,
+        5
+    )
+    expect(resultNftinfo[1].ocean_allocated_realtime_owner).toBeCloseTo(
+        expectedNftinfo[1].ocean_allocated_realtime_owner,
+        5
+    )
```

Comment:
I think I got this
1. test floating values w/ error margin
2. drop floating value results from object
3. strict validate all other values

Should you verify that allocationMultiplier was called inside calculateAllocations?
Otherwise, LGTM.